src/d/actor/d_a_pw.cpp

```diff
@@ -107,8 +111,253 @@ void first_mode_change(pw_class*) {
+            case InvisibleAtStart:
+                i_this->m368 = 8;
+                break;
+            default:
+                i_this->m346 = 1;
+                i_this->m368 = 0xD;
+            }
+        }
+        break;
+    case 0xB:
+        cXyz camfwd;
+        camera_class* pCamera = dComIfGp_getCamera(0);
+        dKyr_get_vectle_calc(&i_this->current.pos, &pCamera->mLookat.mCenter, &camfwd);
+        i_this->mActorPlace.pos.x = i_this->m2CC.x + camfwd.x * 150.0f;
+        i_this->mActorPlace.pos.y = i_this->m2CC.y;
+        i_this->mActorPlace.pos.z = i_this->m2CC.z + camfwd.z * 150.0f;
+        fopAc_ac_c* searchResult;
+        if (i_this->m370 != -1 && fopAcM_SearchByID(i_this->m370, &searchResult) &&
+            searchResult != NULL && fpcM_GetName(&searchResult) == 0xC2)
+        {
+            searchResult->current.pos = i_this->m2CC;
+            searchResult->current.angle.y = i_this->shape_angle.GetY();
+            searchResult->scale = cXyz::BaseXYZ;  // Might not be this, but all ones.
+            BOOL weirdBool = TRUE;
+            if(i_this->m2B4->mFrameCtrl.getState() & 1 == 0 && i_this->m2B4->mFrameCtrl.getRate()) {
+                weirdBool = FALSE;
+            }
+            if(weirdBool) {
+                i_this->m341 = 0;
+                i_this->mBehaviorType = VisibleFromStart;
+                first_mode_change(i_this);
+                i_this->m382 = 3;
+                i_this->m346 = 1;
+                i_this->m368 = 0xD;
+            }
+        }
+        break;
+    case 0xD:
+        i_this->m384 = 0;
+        i_this->m388 = 0;
+        i_this->m38C = 0;
+        i_this->m38E = 0;
+        i_this->m340 = 0;
+        i_this->m341 = 0;
+        randomValue = cM_rndF(60.0f);
+        i_this->m378 = (short) randomValue + 60.0f;
+        if(i_this->m346 && i_this->m374 != 0x23) {
+            anm_init(i_this,0x23,7.0f,'\x02', 1.0f, -1);
+        } else if (i_this->m374 != 0x24) {
+            anm_init(i_this,0x24,7.0,'\x02',1.0f, -1);
+        }
+        i_this->m368 += 1;
+    case 0xE:                       // 0.0f is a placeholder for some local value
+        cLib_addCalc0(&i_this->speedF, 0.0f, 1.0f);
+        if(!i_this->m378) {
+            i_this->m368 += 1;
+            PART_OF_CASE_E:
+            randomValue = cM_rndF(120.0f);
+            i_this->m378 = (short) randomValue + 120.0f;
+            if(i_this->m346 && i_this->m374 != 0x14) {
+                anm_init(i_this,0x14, 7.0f, '\x02', 1.0f, -1);
+            } else if (i_this->m374 != 0x15) {
+                anm_init(i_this,0x15, 7.0f, '\x02', 1.0f, -1);
+            }
+            randomValue = cM_rndFX(16384.0f);
+            i_this->m38C += (short) randomValue;
+            if(i_this->mPathIndex == 0xFF || i_this->m348 == NULL) {
+                i_this->m368 = 0x10;
+            } else {
+                i_this->m368 = 0x14;
+            }
+        }
+        break;
+    case 0xF:
+        goto PART_OF_CASE_E;
```

Comment:
FYI this is more likely to be a fallthrough from case E to case F than a goto, I ran into a lot of fallthroughs when doing these massive enemy switch statements before